coding/python_testing.rst

```diff
@@ -42,44 +41,134 @@ The important things to note in this example are:
+If a test command fails, that output is renamed to have a :file:`.failed` extension and is reported by :command:`scons`.
+For the case where :command:`pytest` is running multiple of the test files, there is a single file created in the :file:`tests/.tests` directory, along with an XML file containing the test output in JUnit format.
+For convenience the output from the main :command:`pytest` run (as opposed to the rare standalone usages) is also written to standard output so is visible in the log or in the shell along with other :command:`scons` output.
+
+Common Issues
+=============
 
 `pytest`_ can run tests from more than one file in a single invocation and this can be used to verify that there is no state contaminating later tests.
-To run `pytest`_ use the :command:`py.test` executable:
+To run `pytest`_ use the :command:`pytest` executable:
 
 .. code-block:: shell
 
-   $ py.test
+   $ pytest
 
 to run all files in the ``tests`` directory named ``test_*.py``. To ensure that the order of test execution does not matter it is useful to sometimes run the tests in reverse order by listing the test files manually:
 
 .. code-block:: shell
 
-   $ py.test `ls -r tests/test_*.py`
+   $ pytest `ls -r tests/test_*.py`
+
+.. note::
+
+  `pytest`_ plugins are usually all enabled by default.
+  In particular, if you install the, otherwise excellent, ``pytest-random-order`` plugin to randomize your tests, this will most likely break your builds as it interacts badly with ``pytest-xdist`` used by :command:`scons` when ``-j`` is used.
+  You can install it temporarily for investigative purposes so long as it is uninstalled afterwards.
+
 
 When writing tests it is important that tests are skipped using the proper :mod:`unittest` :ref:`skipping framework <python:unittest-skipping>` rather than returning from the test early.
 :mod:`unittest` supports skipping of individual tests and entire classes using decorators or skip exceptions.
 LSST code sometimes raises skip exceptions in :meth:`~unittest.TestCase.setUp` or :meth:`~unittest.TestCase.setUpClass` class methods.
 It is also possible to indicate that a particular test is expected to fail, being reported as an error if the test unexpectedly passes.
 Expected failures can be used to write test code that triggers a reported bug before the fix to the bug has been implemented and without causing the continuous integration system to die.
-One of the primary advantages of using a modern test runner such as `pytest`_ is that it is very easy to generate machine-readable pass/fail/skip/xfail statistics to see how the system is evolving over time.
+One of the primary advantages of using a modern test runner such as `pytest`_ is that it is very easy to generate machine-readable pass/fail/skip/xfail statistics to see how the system is evolving over time, and it is also easy to enable code coverage.
+Jenkins now provides test result information.
```

Comment:
Reminds me that we need docs (another ticket) for how to use new Jenkins and get results...

Reply:
Not helped by Blue Ocean hiding all the output except the failures and a note of the skipped tests (but not telling you the reason for the skip).